### PR TITLE
eth, consensus/istanbul: improve sending messages

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: eth/istanbul_handler.go [hunk 1]
```diff
@@ -144,13 +143,20 @@ func (pm *istanbulProtocolManager) eventLoop() {
...
-	if p == nil {
-		log.Warn("Failed to find peer by address", "addr", event.Target)
-		return
+	for _, p := range pm.peers.Peers() {
+		pubKey, err := p.ID().Pubkey()
+		if err != nil {
+			continue
+		}
+		addr := crypto.PubkeyToAddress(*pubKey)
+		if event.Targets[addr] == true {
```
Comment (bailantaotao) [inline eth/istanbul_handler.go]: event.Targets[addr] is enough

File: consensus/istanbul/backend/backend.go [hunk 2]
```diff
@@ -99,16 +99,9 @@ func (sb *backend) Validators(proposal istanbul.Proposal) istanbul.ValidatorSet
...
-	go sb.eventMux.Post(istanbul.ConsensusDataEvent{
-		Target: target,
-		Data:   payload,
-	})
-	return nil
-}
-
 // Broadcast implements istanbul.Backend.Send
 func (sb *backend) Broadcast(valSet istanbul.ValidatorSet, payload []byte) error {
+	targets := make(map[common.Address]bool)
```
Comment (bailantaotao) [inline consensus/istanbul/backend/backend.go]: Why do you use map here? instead of array?

File: eth/istanbul_handler.go (hunk 1 above)
Comment (markya0616) [inline eth/istanbul_handler.go]: OK

File: consensus/istanbul/backend/backend.go (hunk 2 above)
Comment (markya0616) [inline consensus/istanbul/backend/backend.go]: map is faster than an array if we want to look up it

**Pull request merged**